src/Utils/util.cpp

```diff
@@ -118,14 +118,30 @@ StackValue *readWasmArgs(Type function, uint8_t *data) {
     return args;
 }
 
-bool deserialiseStackValue(uint8_t *input, StackValue *value) {
-    uint8_t valtypes[] = {I32, I64, F32, F64};
-    uint8_t type_index = *input++;
-    if (type_index >= sizeof(valtypes)) return false;
-    value->value.uint64 = 0;  // init whole union to 0
-    size_t qb = type_index == 0 || type_index == 2 ? 4 : 8;
-    value->value_type = valtypes[type_index];
-    memcpy(&value->value, input, qb);
+bool deserialiseStackValue(uint8_t *input, bool decodeType, StackValue *value) {
+    if (decodeType) {
+        uint8_t valtypes[] = {I32, I64, F32, F64};
+        uint8_t type_index = *input++;
+        if (type_index >= sizeof(valtypes)) return false;
+        value->value.uint64 = 0;  // init whole union to 0
+        value->value_type = valtypes[type_index];
+    }
```

Comment:
It is indeed, to pass the type along with the new value. It is mainly meant to reduce code duplication. When implementing this https://github.com/TOPLLab/WARDuino/issues/127, I will have to reimplement the *interruptRecvState* handler to account for virtual addresses. And since interruptRecvState receives new stackvalues and Globals. It is thus then perfect opportunity for reducing code duplication by using this serialiseStackValue.


In retrospect we could remove the possibility of changing the the type of a stackvalue when using interruptUpdateStackValue because it makes no sense here. But we will definitely need it for the interruptRecvState.

Reply:
Thanks for the clarifications, that makes sense.

> In retrospect we could remove the possibility of changing the the type of a stackvalue when using interruptUpdateStackValue because it makes no sense here.

I agree, and it avoids a lot of issues where the stack values have the wrong type for function calls and instructions. I suggest we change that for this PR already.

Reply:
I can do it

Reply:
Latest commit solves the issue a668fd5bcd953471a9b540c07cfc0a6cb5a3910a